influxdata/influxdb_iox
# chore: Update datafusion

# Rationale:
1. Get fix for https://github.com/apache/arrow-datafusion/pull/1047 (FYI @NGA-TRAN)

# Changes:
1. Update to use latest datafusion
2. run `cargo update` and check in results
2. Update to latest sqlparser, remove unused dependencies
3. Enable tests that were failing due to https://github.com/apache/arrow-datafusion/pull/1047 (changes in https://github.com/influxdata/influxdb_iox/pull/2618)
4. Update explain plan output due improvements in display from  https://github.com/apache/arrow-datafusion/pull/971  @matthewmturner 



# Other crates that are updated:

```
alamb@ip-192-168-182-158 influxdb_iox % cargo update
...
cargo update
    Updating crates.io index
    Updating git repository `https://github.com/apache/arrow-datafusion.git`
    Updating git repository `https://github.com/Azure/azure-sdk-for-rust.git`
    Updating git repository `https://github.com/mkmik/heappy`
    Updating instant v0.1.10 -> v0.1.11
    Updating openssl-sys v0.9.66 -> v0.9.67
    Updating pkg-config v0.3.19 -> v0.3.20
    Updating syn v1.0.76 -> v1.0.77
    Updating tinyvec v1.4.0 -> v1.5.0
    Updating zeroize v1.4.1 -> v1.4.2
```
